src/sage/arith/misc.py

```diff
@@ -1027,10 +1027,11 @@ def primes(start, stop=None, proof=None):
 
     INPUT:
 
-    - ``start`` - an integer - lower bound for the primes
+    - ``start`` - an integer (default: 2) optional argument -
+      giving lower bound for the primes
 
-    - ``stop`` - an integer (or infinity) optional argument -
-      giving upper (open) bound for the primes
+    - ``stop`` - an integer (or infinity) - giving upper (open) bound for the
+      primes
```

Comment:
I see your point mathematically. On the other hand, ``stop`` is actually an optional argument in the code sense. Should you change the argument to the function rather than changing the docs?

Then see [this SO post](https://stackoverflow.com/questions/50540876/python-optional-parameters-before-the-required-ones) for keeping the user interface the same.

Reply:
Should I change the signature to `primes(start=2, stop=None, proof=None)` ?

Reply:
Thats what I was thinking.